analyses/sv-analysis/01-process-sv-file.R

```diff
@@ -0,0 +1,187 @@
+      } else {
+        dels_file[id_name, "inserlen"] <- 0
+      }
+      dels_file[id_name, "strand1"] = "+"
+      dels_file[id_name, "strand2"] = "-"
+    }
+  }
+  ## TYPE == INV
+  invs <- sv_individual[sv_individual$SV.type == "INV", ]
+  reads_id_inv <- substr(invs$ID, start = 1, stop = nchar(as.character(invs$ID)))
+  invs_file <- matrix(NA, nrow = length(unique(reads_id_inv)), ncol = 11)
+  colnames(invs_file) <- c("chrom1","pos1","chrom2","pos2","SVtype","strand1","strand2","alt1","alt2","homolen","inserlen")
+  rownames(invs_file) <- unique(reads_id_inv)
+  if (nrow(invs)  != 0) {
+    for (j in 1:nrow(invs)) { 
+      id_name <- substr(invs[j, 7], start = 1, stop = nchar(as.character(invs[j, 7])))
+      invs_file[id_name, "chrom1"] <- as.character(invs$SV.chrom[j])
+      invs_file[id_name, "chrom2"] <- as.character(invs$SV.chrom[j])
+      invs_file[id_name, "pos1"] <- as.character(invs$SV.start[j])
+      invs_file[id_name, "pos2"] <- as.character(invs$SV.end[j])
+      invs_file[id_name, "SVtype"] <- "INV"
+      invs_file[id_name, "alt1"] <- as.character(invs$AnnotSV.ID[j])
+      if (str_detect(as.character(invs$INFO[j]),"HOMLEN=")) {
+        invs_file[id_name, "homolen"] <- gsub((".*HOMLEN="),"",as.character(invs$INFO[j]))
+        invs_file[id_name, "homolen"] <- gsub((";.*"),"",invs_file[id_name, "homolen"])
+      } else {
+        invs_file[id_name, "homolen"] <- 0
+      }
+      if (str_detect(as.character(invs$INFO[j]),"SVINSLEN=")) {
+        invs_file[id_name, "inserlen"] <- gsub((".*SVINSLEN="),"",as.character(invs$INFO[j]))
+        invs_file[id_name, "inserlen"] <- gsub((";.*"),"",invs_file[id_name, "inserlen"])
+      } else {
+        invs_file[id_name, "inserlen"] <- 0
+      }
+      if (str_detect(invs$INFO[j],"INV5")) {
+        invs_file[id_name, "strand1"] = "+"
+        invs_file[id_name, "strand2"] = "+"
+        invs_file[id_name, "SVtype"] <- "h2hINV"
+      } else {
+        invs_file[id_name, "strand1"] = "-"
+        invs_file[id_name, "strand2"] = "-"
+        invs_file[id_name, "SVtype"] <- "t2tINV"
+      }
+    }
+  }
+  
+  ## Merge four types SV
+  sv_merge <- rbind(dups_file, dels_file, invs_file, bnds_file)
+  sv_merge <- sv_merge[is.na(sv_merge[, 1]) == FALSE & is.na(sv_merge[, 3]) == FALSE, ]
+  outputname_sv_merge <- paste(i,".vcf",sep = "")
+  
+  ## Shatterseek only accept file with chrom1-22,X, so remove chrY and ChrM
+  sv_merge_withoutY_withoutM <- sv_merge[sv_merge[,1] != "Y" & sv_merge[,3] != "Y" & sv_merge[,1] != "M" & sv_merge[,3] != "M",]
+  outputname_sv_merge_withoutY_withoutM <- paste(i,"_withoutYandM.vcf",sep = "")
+  
+  ## output files
+  write.table(sv_merge,file = outputname_sv_merge,sep = "\t",quote = FALSE,row.names = TRUE,col.names = NA)
+  write.table(sv_merge_withoutY_withoutM,file = outputname_sv_merge_withoutY_withoutM,sep = "\t",quote = FALSE,row.names = TRUE,col.names = NA)
```

Comment:
I was running this locally and noticed that these files were being written in the `analyses/sv-analysis` directory. We generally do not want to track these files in git, so we can make it a little easier for others to work with this code by either explicitly ignoring them or by putting them in the `scratch` directory. 

Here's the blurb from the [Folder Structure](https://github.com/AlexsLemonade/OpenPBTA-analysis#folder-structure) section of the README:

> Files that are primarily tabular results files should be placed in a `results` subdirectory. Intermediate files that are useful within the processing steps but that do not represent final results should be placed in `../../scratch/`.

So the options are 

#### `results` and `.gitignore`

You could create a `results` directory in this folder (`analyses/sv-analysis`), assuming you are using the `root_dir` suggestion above, with something like

```r
results_directory <-  file.path(root_dir, "analyses/sv-analysis/results")
if (!dir.exists(results_directory)) {
  dir.create(results_directory, recursive = TRUE)
}
```  

at the beginning of the script. The idea being that if the results directory doesn't exist yet, the script will create it to avoid any trouble writing files once you get here. 

Then you can add a text file called `.gitignore` to the `analyses/sv-analysis` directory that tells git to ignore your vcf files with 
```
# ignore all files with the vcf extension in this analysis module
*.vcf
```

Here's an example of someone using a `.gitignore` file specifically for their analysis module: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/54a8a1ae9d91881aed0360df1ac47df8d940c139/analyses/sample-distribution-analysis/.gitignore and some GitHub documentation: https://help.github.com/en/github/using-git/ignoring-files

Then when you write these files you would use `file.path(results_directory, outputname_sv_merge)` 

#### `scratch`

As you mentioned you could put all your output files in `scratch`. My personal preference would be to create a subdirectory to hold all the output files using `root_dir` again:

```r
output_directory <- file.path(root_dir, "scratch/sv-vcf")
if (!dir.exists(output_directory)) {
  dir.create(output_directory, recursive = TRUE)
}
```
But you could just use `scratch`. I would still use `root_dir` because that will allow you and other contributors to run this from anywhere within the project.

Reply:
Aha, I think I understand. 
`rprojroot` sounds a great idea! I will try it soon.
As for the output path, I would like to use  `scratch`, it looks much easier.
Thank you so much @jaclyn-taroni

Reply:
Sounds good 👍  Happy to help, let us know if you have any questions as you are implementing the `rprojroot` change.